### PR TITLE
Remove implicit casts from {integer,float} to bool.

### PR DESCRIPTION
**Pull request by petermattis**

### CODE REVIEW

**Comment by petermattis** [comment]
Grr, the `hub` tool gave this PR the wrong title initially.

**Comment by tamird** [comment]
No test changes? LGTM

**Comment by petermattis** [comment]
We weren't actually testing the implicit conversions to bool before. I'll add some failure cases.

File: sql/parser2/eval_test.go [hunk 1]
```diff
@@ -169,6 +169,10 @@ func TestEvalExprError(t *testing.T) {
 		{`'10' > 2`, `unsupported comparison operator:`},
 		{`1 IN ('a', 'b')`, `unsupported comparison operator:`},
 		{`a`, `column \"a\" not found`},
+		{`1 AND true`, `cannot convert int to bool`},
```
Comment (tamird) [inline sql/parser2/eval_test.go]: :+1:

**Comment by tschottdorf** [comment]
LGTM

**Pull request merged**